platform/src/main/java/org/hillview/storage/FileSetDescription.java

```diff
@@ -71,20 +71,21 @@
     public Double startTime;
     @Nullable
     public Double endTime;
-
+    
     @Nullable
     private String getSchemaPath() {
         if (Utilities.isNullOrEmpty(this.schemaFile))
             return null;
-        return Paths.get(this.getFolder(), this.schemaFile).toString();
+        return Paths.get(this.getFolder(this.fileNamePattern), this.schemaFile).toString();
     }
 
-    public String getFolder() {
-        int last = this.fileNamePattern.lastIndexOf('/');
-        last = Math.max(last, this.fileNamePattern.lastIndexOf('\\'));
-        if (last == -1)
+    public String getFolder(String path) {
```

Comment:
Now you can make these methods static. Perhaps they should be moved to the Utilities.